Refactor amount handling to use AssetData

gemdev111 opened the pull request:
Replaces AmountService with direct usage of AssetData and AssetDataRequest for price and balance retrieval in transfer flows. Updates navigation, view models, and factories to remove AmountService dependency, streamlining asset price and balance observation. Adds new primitives and store request types for asset balance and price. Cleans up related tests and navigation stack initializations.

closes #1034

gemdev111 committed:
Refactor amount handling to use AssetBalancePrice

Replaces AmountService with direct usage of AssetBalancePrice and AssetBalancePriceRequest for price and balance retrieval in transfer flows. Updates navigation, view models, and factories to remove AmountService dependency, streamlining asset price and balance observation. Adds new primitives and store request types for asset balance and price. Cleans up related tests and navigation stack initializations.

gemdev111 committed:
Update AssetBalancePriceRequest.swift

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
Updates AssetSceneViewModelTests

Removed unused assetid from AssetBalancePrice, since we have id in price

gemdev111 committed:
Removed stakeservice

gemdev111 committed:
Update JobRunnerTests.swift

gemdev111 committed:
Merge branch 'main'

gemdev111 committed:
Update AmountSceneViewModel.swift

gemdev111 committed:
Updates tests

gemdev111 committed:
Merge branch 'main' into balance-grdb

DRadmir commented on Features/Transfer/Sources/ViewModels/AmountSceneViewModel.swift:
```diff
@@ -309,6 +310,20 @@ extension AmountSceneViewModel {
+        case .smartChain:
+            return StakeHub.address
+        default:
+            fatalError()
```
Can we avoid to using fatalError here? It can cause a crash in production

gemdev111 replied:
it's taken from stakeservice, to remove stakeservice from the module. But yeah it can be improved

DRadmir commented on Features/Transfer/Sources/ViewModels/AmountSceneViewModel.swift:
```diff
@@ -309,6 +310,20 @@ extension AmountSceneViewModel {
     }
 
+    private func getRecipientAddress(chain: StakeChain?, type: AmountType, validatorId: String?) -> String? {
+        guard let id = validatorId else {
```
Move this guard into the specific case, because for smartChain it can cause an early return

DRadmir commented on Packages/Store/Sources/Requests/AssetBalancePriceRequest.swift:
```diff
@@ -0,0 +1,45 @@
+            return nil
+        }
+
+        return AssetBalancePrice(
```
It would be better to use our mapping style. Create an extension for BalancePriceInfo with a map method

gemdev111 committed:
Minor updates

Improve getRecipientAddress in AmountSceneViewModel
Improve AssetBalancePriceRequest for app consistent style

gemcoder21 commented on Packages/Store/Sources/Types/BalancePriceInfo.swift:
```diff
@@ -0,0 +1,20 @@
+
+struct BalancePriceInfo: FetchableRecord, Codable {
+    var asset: AssetRecord
+    var balance: BalanceRecord?
```
```suggestion
    var balance: BalanceRecord
```

gemcoder21 commented on Packages/Store/Sources/Requests/AssetBalancePriceRequest.swift:
```diff
@@ -0,0 +1,32 @@
+import Combine
+import Primitives
+
+public struct AssetBalancePriceRequest: ValueObservationQueryable {
```
Use `AssetRequest`

gemdev111 committed:
Refactor asset balance and price handling in transfer feature

Replaces AssetBalancePrice and related request/types with AssetData and AssetRequest in AmountSceneViewModel and related views/tests.
Removes obsolete AssetBalancePrice, AssetBalancePriceRequest, and BalancePriceInfo files. Updates usages to reflect new data model for asset balance and price.
Minor test timing adjustments in JobRunnerTests.

gemdev111 committed:
Merge branch 'main' into balance-grdb

gemdev111 committed:
Update core

gemdev111 committed:
Update StakeService+TestKit.swift

gemdev111 committed:
Update AmountSceneViewModel.swift
